tests/IResearch/IResearchAqlAnalyzer-test.cpp

```diff
@@ -0,0 +1,574 @@
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Andrei Lobov
+////////////////////////////////////////////////////////////////////////////////
+
+#include "IResearchQueryCommon.h"
+
+
+#include "Aql/AqlFunctionFeature.h"
+#include "IResearch/IResearchView.h"
+#include "Transaction/StandaloneContext.h"
+#include "Utils/OperationOptions.h"
+#include "Utils/SingleCollectionTransaction.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/ManagedDocumentResult.h"
+
+#include <velocypack/Iterator.h>
+#include "IResearch/IResearchAqlAnalyzer.h"
+
+
+class IResearchAqlAnalyzerTest : public IResearchQueryTest {};
+
+namespace {
+constexpr irs::string_ref AQL_ANALYZER_NAME{"aql"};
+
+struct analyzer_token {
+  irs::string_ref value;
+  uint32_t pos;
+};
+
+using analyzer_tokens = std::vector<analyzer_token>;
+
+void assert_analyzer(irs::analysis::analyzer* analyzer, const std::string& data,
+                     const analyzer_tokens& expected_tokens) {
+  SCOPED_TRACE(data);
+  auto* term = irs::get<irs::term_attribute>(*analyzer);
+  ASSERT_TRUE(term);
+  auto* inc = irs::get<irs::increment>(*analyzer);
+  ASSERT_TRUE(inc);
+  ASSERT_TRUE(analyzer->reset(data));
+  uint32_t pos{irs::integer_traits<uint32_t>::const_max};
+  auto expected_token = expected_tokens.begin();
+  while (analyzer->next()) {
+    auto term_value =
+        std::string(irs::ref_cast<char>(term->value).c_str(), term->value.size());
+    SCOPED_TRACE(testing::Message("Term:") << term_value);
+    auto old_pos = pos;
+    pos += inc->value;
+    ASSERT_NE(expected_token, expected_tokens.end());
+    ASSERT_EQ(irs::ref_cast<irs::byte_type>(expected_token->value), term->value);
+    ASSERT_EQ(expected_token->pos, pos);
+    ++expected_token;
+  }
+  ASSERT_EQ(expected_token, expected_tokens.end());
+}
+} // namespace
+
+TEST_F(IResearchAqlAnalyzerTest, test_create_valid) {
+  // const value
+  {
+    auto ptr = irs::analysis::analyzers::get(AQL_ANALYZER_NAME,
+                                             irs::type<irs::text_format::vpack>::get(),
+                                             arangodb::iresearch::ref<char>(
+                                               VPackParser::fromJson("{\"queryString\": \"RETURN '1'\"}")->slice()),
+                                             false);
+    ASSERT_NE(nullptr, ptr);
+    assert_analyzer(ptr.get(), "2", {{"1", 0}});
+  }
+  // just parameter
+  {
```

Comment:
Can we please put these test blocks into single `TEST_F` cases?
We have seen that one large TEST_F with many of them takes ages to compile.
Also if only of them does not work, you only see one at a time, you do not see if it is only one, or all of them that do not work.

Reply:
Currently all ArangoSearch tests follows this pattern. If this pattern gives measurable complile/link performance degradation it worth changing it everywhere and by separate PR,

Reply:
Yes it does, and yes I know ;)

The tests that take longest to compile all follow this pattern and splitting it up as I described helped to speed up by a significant factor (got factor ~100 on one of them)